Sources/UIScreen.swift

```diff
@@ -10,16 +10,18 @@ import SDL
 import SDL_gpu
 import func Foundation.round
 
-
 extension SDLWindowFlags: OptionSet {}
 
 public extension UIScreen {
-    // Default value for testing. Gets set in TestSetup.swift
-    internal(set) public static var main: UIScreen!
+    internal(set) public static var main: UIScreen! {
+        didSet { CALayer.layerTreeIsDirty = true }
```

Comment:
I ask myself more often if CAlayer is the right namespace for `layerTreeIsDirty`. Maybe it would be clearer if we move it to UIScreen? ... proably itchy

Reply:
Yes I wondered the same but then it's weird for random CALayers to access `UIScreen.main` too. The best option might be to do this via proxy: e.g.

```
// in a layer:
CALayer.invalidateLayerTree()

extension CALayer {
  func invalidateLayerTree() { 
     UIScreen.main?.needsRerender = true
  }
}
```

What do you think about that solution? @michaelknoch

Reply:
I don't think accessing UIScreen.main from here is that random. Its more random for me that CALayer has a static context at all. I always think layerTreeIsDirty is an instance property and the layer and all its childLayers is dirty. Its kind of stupid because it obviously is a static property but I had this confusion several times already ^^

Reply:
It would be more correct to bubble changes up through the layer tree:

```
extension CALayer {
  private var isDirty = true
  func invalidate() {
    isDirty = true
    superlayer?.invalidate()
  }
}
```

But we don't need the granularity and this would have a high performance cost for no benefit over the current solution (because if one thing is dirty we have to rerender the entire tree anyway)

Reply:
It wouldn't be correct to access `UIScreen.main` from random layers, because there's nothing to say that layer has any association with that specific screen at all. And if we end up supporting multiple "screens" we'd have to do a more major refactor than what we'd have to now.

Thinking about this further, this could make the most sense as a property on `UIApplication.shared.keyWindow`, but again I don't think we have a reasonable way of accessing a view's associated window (yet). Also the window does not get destroyed alongside the GLContext as is, meaning we wouldn't gain much from having this property on the window instance anyway (it would have to be treated as if it was static anyway)

So I think I'm just going to leave this for now. But I agree that it's still an unanswered question in our design.